src/instruments/src/PFD/FMA.tsx

```diff
@@ -1,554 +1,1021 @@
+                this.inModeReversionPathRef.instance.setAttribute('visibility', 'hidden');
+                this.boxClassSub.set('NormalStroke White');
+                this.displayModeChangedPath(true);
+            }
+        });
+
+        sub.on('fmaSpeedProtection').whenChanged().handle((protection) => {
+            this.inSpeedProtection = protection;
+            if (!protection) {
+                this.displayModeChangedPath(true);
+                this.speedProtectionPathRef.instance.setAttribute('visibility', 'hidden');
+            } else {
+                this.displayModeChangedPath();
+                this.speedProtectionPathRef.instance.setAttribute('visibility', 'visible');
+            }
+            this.getText();
+        });
+
+        sub.on('expediteMode').whenChanged().handle((e) => {
+            this.expediteMode = e;
+            this.getText();
+        });
+    }
+
+    render(): VNode {
+        return (
+            <g>
+
+                <path ref={this.modeChangedPathRef} class={this.boxClassSub} visibility="hidden" d={this.boxPathStringSub} />
+
+                <path ref={this.speedProtectionPathRef} class="NormalStroke Amber BlinkInfinite" d="m34.656 1.8143h29.918v6.0476h-29.918z" />
+                <path ref={this.inModeReversionPathRef} class="NormalStroke White BlinkInfinite" d="m34.656 1.8143h29.918v6.0476h-29.918z" />
+
+                <text ref={this.fmaTextRef} style="white-space: pre" class="FontMedium MiddleAlign Green" x="49.921795" y="7.1040988">
+
+                    {/* set directly via innerhtml as tspan was invisble for some reason when set here */}
+
+                </text>
+            </g>
         );
-        break;
     }
-    default:
-        return null;
+}
+
+class B2Cell extends DisplayComponent<CellProps> {
+    private text1Sub = Subject.create('');
+
+    private text2Sub = Subject.create('');
+
+    private classSub = Subject.create('');
+
+    onAfterRender(node: VNode): void {
+        super.onAfterRender(node);
+
+        const sub = this.props.bus.getSubscriber<PFDSimvars>();
+
+        sub.on('fmaVerticalArmed').whenChanged().handle((fmv) => {
+            const altArmed = (fmv >> 0) & 1;
+            const altCstArmed = (fmv >> 1) & 1;
+            const clbArmed = (fmv >> 2) & 1;
+            const desArmed = (fmv >> 3) & 1;
+            const gsArmed = (fmv >> 4) & 1;
+            const finalArmed = (fmv >> 5) & 1;
+
+            let text1: string;
+            let color1 = 'Cyan';
+            if (clbArmed) {
+                text1 = 'CLB';
+            } else if (desArmed) {
+                text1 = 'DES';
+            } else if (altCstArmed) {
+                text1 = 'ALT';
+                color1 = 'Magenta';
+            } else if (altArmed) {
+                text1 = 'ALT';
+            } else {
+                text1 = '';
+            }
+
+            let text2;
+            // case 1:
+            //     text2 = 'F-G/S';
+            //     break;
```

Comment:
Same not as above applies here - we might as well enable this?

Reply:
This one is a bit special, it was also copied but there seems no switch-case statement where it could belong to